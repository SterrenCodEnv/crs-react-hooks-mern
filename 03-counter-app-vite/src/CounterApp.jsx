import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleIncrease = () => {
    setCounter(counter + 1);
  };

  const handleDecrease = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>

      <button onClick={handleIncrease}> +1 </button>
      <button onClick={handleDecrease}> -1 </button>
      <button aria-label="btn-reset" onClick={handleReset}>
        {' '}
        Reset{' '}
      </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
