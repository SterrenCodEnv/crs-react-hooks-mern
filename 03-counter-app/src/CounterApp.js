import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({year = 2021}) => {

    const [counter, setCounter] = useState(0);
    const handleAdd = () => setCounter((c) => c + 1) ;
    const handleSubstract = () => setCounter((c) => c - 1) ;
    const handleReset = () => setCounter((c) => c = 0);
    
    return (
    <>
        <h1>CounterApp - {year}</h1>
        <h2> { counter } </h2>
        <button onClick={ handleAdd }>+1</button>
        <button onClick={ handleReset }>Reset</button>
        <button onClick={ handleSubstract }>-1</button>
    </>
    );
}

CounterApp.propTypes = {
    year : PropTypes.number.isRequired
}

export default CounterApp;