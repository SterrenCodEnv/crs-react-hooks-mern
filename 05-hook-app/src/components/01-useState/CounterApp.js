import React, { useState } from "react";
import "./counter.css";

export const CounterApp = () => {
  const [state, setState] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
    counter4: 40,
  });

  const {counter1, counter2} = state;

  return (
    <div>
      <h2>CounterOne {counter1}</h2>
      <h2>CounterTwo {counter2}</h2>

      <hr />
      <button
        onClick={() => {
            setState({...state, counter1: counter1 + 1});
        }}
        className="btn btn-primary"
      >
        +1
      </button>
    </div>
  );
};
