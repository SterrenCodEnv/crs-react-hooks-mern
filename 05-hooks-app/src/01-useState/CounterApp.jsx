import { useState } from 'react';

export const CounterApp = () => {
  const [counter, setCounter] = useState(0);

  const handleIncrease = () => {
    setCounter(counter + 1);
  };

  const handleDecrement = () => {
    counter > 0 && setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(0);
  };

  return (
    <>
      <h2>CounterApp</h2>
      <hr />
      <h3>Counter: {counter}</h3>
      <button className="btn btn-success" onClick={handleIncrease}>
        +1
      </button>
      <button className="btn btn-danger" onClick={handleReset}>
        Reset
      </button>
      <button className="btn btn-warning" onClick={handleDecrement}>
        -1
      </button>
    </>
  );
};
