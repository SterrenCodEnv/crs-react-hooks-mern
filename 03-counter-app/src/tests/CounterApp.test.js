import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CounterApp from '../CounterApp';


describe('Test in CounterApp.js', () => {
    
    // Intelligence functions persist
    let wrapper = shallow( <CounterApp year={2021} />);
    beforeEach(() => {
        wrapper = shallow( <CounterApp year={2021} /> )
    });

    test('Show year app', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("The title is 'CounterApp - 2021'", () => {
        const year = 2021
        const wrapper = shallow(<CounterApp year={year}/>)
        const titleText = wrapper.find('h1').text();
        expect(titleText).toBe(`CounterApp - ${year}`);
    })

    test('Verify handleAdd function in add button', () => {
        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('1');
    });

    test('Verify handleSubstract function in substract button', () => {
        wrapper.find('button').at(2).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('-1');
    });

    test('Verify handleReset function in reset button', () => {
        wrapper.find('button').at(1).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('0');
    });

});