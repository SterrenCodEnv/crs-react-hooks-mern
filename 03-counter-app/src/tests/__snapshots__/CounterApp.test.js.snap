// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in CounterApp.js Show year app 1`] = `
<Fragment>
  <h1>
    CounterApp - 
    2021
  </h1>
  <h2>
     
    0
     
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
